Added support of line alignment on CanvasRenderer

SerG-Y opened the pull request:
##### Description of change
<!-- Provide a description of the change below this comment. -->
Implementation of #7130 

Example: 
https://jsfiddle.net/SerG_Y/qycon5Lf/

##### Pre-Merge Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

SukantPal commented:
This fails with a line alignment of 0.75, for example (middle):

<img width="515" alt="Screen Shot 2021-01-15 at 9 35 00 AM" src="https://user-images.githubusercontent.com/22450567/104739645-f23d5400-5714-11eb-95a9-adac814fd49e.png">

SukantPal commented on packages/canvas/canvas-graphics/src/CanvasGraphicsRenderer.ts:
```diff
@@ -73,6 +73,118 @@ export class CanvasGraphicsRenderer
+            {
+                context.save();
+                context.rect(rectangle.x, rectangle.y, rectangle.width, rectangle.height);
+                context.clip();
```
Instead of using clipping for rectangles, how about we draw two rectangles and the `evenodd` fill rule to do a stroke? That would be faster since the Canvas2D implementation won't have to do clipping calculations.

SukantPal commented on packages/canvas/canvas-graphics/src/CanvasGraphicsRenderer.ts:
```diff
@@ -73,6 +73,118 @@ export class CanvasGraphicsRenderer
+     * @param {string|CanvasPattern} contextStrokeStyle
+     * @param {number} worldAlpha
+     */
+    private draw(
```
I dunno if we should call this `paint` (fill + stroke = paint) since `draw` is for actually pushing a primitive/shape into the rendering pipeline (cc @bigtimebuddy)

bigtimebuddy replied:
It's private, so it doesn't matter to me. If it were public, I'd scrutinize this more. `paint` is fine too.

SerG-Y commented:
Hey @SukantPal, @bigtimebuddy
I am sorry, it's my fail because I considered only support alignment exactly like [0, 0.5, 1]. Should we support alignment only in the range [0-1] or it can be any number?

SukantPal commented:
It can be any number. Usually in (0-1) range

bigtimebuddy commented:
I appreciate how clean this PR is @SerG-Y. Awesome work. My biggest request is the float 0-1. I think for numbers outside that range, we can clamp or consider it undefined behavior.

bigtimebuddy commented:
I like this but there are some limitations, for example what about polygons? For example, `drawStar` is a closed shape that uses `drawPolygon`.

https://jsfiddle.net/bigtimebuddy/9ga2kod0/

SerG-Y commented:
Hey @bigtimebuddy, thanks for the review, I appreciate it. It's still in progress. I know about this limitation and I am working on it, hope will be able to deliver it till the end of the week. Also, it has some glitches (rounded rectangle, ellipse) when stroke width bigger than the radius, hope I will be able to fix it as well. Thanks! 
<img width="785" alt="Screen Shot 2021-01-25 at 13 42 56" src="https://user-images.githubusercontent.com/12532719/105701736-6e087f00-5f13-11eb-8d4b-b81786b82c48.png">

SerG-Y commented:
Hey @bigtimebuddy, sorry for so long delay, It a bit harder than I thought for the first time (just support 3 numbers [0, 0.5, 1])😄. I updated my PR, added line alignment support for polygon and fixed some glitches.

I have one question, I see that for polygon in WebGL we not consider is polygon clockwise or not, should we consider it?
When `alignment=1` but stroke positioned inside the shape
Example:
https://jsfiddle.net/SerG_Y/7L3svu8t/

PR examples:
1) Shapes -  https://jsfiddle.net/SerG_Y/qycon5Lf/
2) Polygon - https://jsfiddle.net/SerG_Y/vjgz129b/

SukantPal commented:
Serhiy, I think we had remember that we decided against checking whether a polygon was clockwise or not for alignment. That’s because it would be impossible to determine the inside/outside of an open polyline. Other thoughts welcome on this.

bigtimebuddy commented:
Clearly the clockwise bias is still in there for WebGL. I think this is a bug for WebGL and I agree with @SukantPal that we should ignore the orientation. I think addressing that is outside the scope of this PR. We should be trying to do in WebGL what you're doing with canvas here, not the other way around.

SerG-Y commented:
Hey @bigtimebuddy, @SukantPal will you be able to review it? Maybe I can help somehow (additional examples etc.)?

SukantPal commented:
Things have slowed a little on the PRs. I'll try to get a review in the next 36 hours. Even my #7213 PR needs more eyes :)

SukantPal reviewed (approved):
👍 Let's see how this does in the wild!

bigtimebuddy commented:
Thanks @SerG-Y for your hard work on this and for handling all the feedback.

SerG-Y commented:
@ivanpopelyshev Thanks, man!
